src/server/utils/devCdnFactory.js

```diff
@@ -0,0 +1,199 @@
+
+      const parsedBundlePath = path.parse(module.node.url, '');
+
+      // store urls for later lookup when module requests are caught by one-app-dev-cdn
+      remoteModuleBaseUrls.push(
+        module.node.url.replace(parsedBundlePath.base, '')
+      );
+
+      // override remote module map to point all module URLs to one-app-dev-cdn
+      module.node.url = module.node.url.replace(
+        new URL(module.node.url).origin, oneAppDevStaticsAddress
+      );
+      module.legacyBrowser.url = module.legacyBrowser.url.replace(
+        new URL(module.legacyBrowser.url).origin, oneAppDevStaticsAddress
+      );
+      module.browser.url = module.browser.url.replace(
+        new URL(module.browser.url).origin, oneAppDevStaticsAddress
+      );
+      return module;
+    });
+    return remoteModuleMap;
+  } catch (error) {
+    console.warn(
+      `one-app-dev-cdn error loading module map from ${remoteModuleMapUrl}: ${error}`
+    );
+    return {};
+  }
+};
+
+export const oneAppDevCdnFactory = ({
+  localDevPublicPath,
+  remoteModuleMapUrl,
+  useLocalModules,
+  appPort,
+  useHost,
+  routePrefix = '',
+}) => {
+  if (!remoteModuleMapUrl && !useLocalModules) {
+    throw new Error('remoteModuleMapUrl is a required param when useLocalModules is not true');
+  }
+  if (!localDevPublicPath) { throw new Error('localDevPublicPath is a required param'); }
+  if (!appPort) { throw new Error('appPort is a required param'); }
+
+  if (remoteModuleMapUrl) {
+    console.log(`one-app-dev-cdn loading module map from ${remoteModuleMapUrl}`);
+  } else {
+    console.log('one-app-dev-cdn only using locally served modules');
+  }
+
+  const oneAppDevCdn = Fastify();
+
+  if (process.env.NODE_ENV === 'production') {
+    console.warn('do not include one-app-dev-cdn in production');
+    return oneAppDevCdn;
+  }
+  oneAppDevCdn.register(compress, {
+    global: false,
+  });
+  oneAppDevCdn.register(cors, {
+    origin: [
+      `http://localhost:${appPort}`,
+      `http://${ip.address()}:${appPort}`,
+      undefined,
+    ],
+  });
+
+  // for locally served modules
+
+  oneAppDevCdn.register(fastifyStatic, {
+    root: `${localDevPublicPath}/modules`,
+    prefix: `${routePrefix}/modules`,
+    index: false,
+
+  });
+
+  const remoteModuleBaseUrls = [];
+  // support one-app-cli's "serve-module"
+  // merge local with remote, with local taking preference
+  oneAppDevCdn.get(`${routePrefix}/module-map.json`, async (req, reply) => {
+    const hostAddress = useHost ? `http://${req.headers.host}` : `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}`;
+
+    const localMap = useLocalModules ? JSON.parse(getLocalModuleMap({
+      pathToModuleMap: path.join(localDevPublicPath, 'module-map.json'),
+      oneAppDevCdnAddress: hostAddress,
+    })) : {};
+    let remoteMap = {};
+
+    if (remoteModuleMapUrl !== undefined) {
+      remoteMap = await consumeRemoteRequest(remoteModuleMapUrl, hostAddress, remoteModuleBaseUrls);
+    }
```

Comment:
we can simplify it by just converting it into an inline condition statement, e.g.

```
const remoteMap = remoteModuleMapUrl != null ? await consumeRemoteRequest(remoteModuleMapUrl, hostAddress, remoteModuleBaseUrls) : {}
```

Mutation makes code less readable because we have to search for the parts where the mutation happens, in this case, it can be avoided